cli_tools/gce_ovf_import/ovf_importer/ovf_param_validator.go

```diff
@@ -0,0 +1,255 @@
+	zoneSplits := strings.Split(zone, "-")
+	if len(zoneSplits) < 2 {
+		return "", daisy.Errf("%v is not a valid zone", zone)
+	}
+	return strings.Join(zoneSplits[:len(zoneSplits)-1], "-"), nil
+}
+
+func (p *ParamValidatorAndPopulator) determineZone(project, originalZone string) (zone string, err error) {
+	zone = strings.TrimSpace(originalZone)
+	if zone != "" {
+		if err := p.zoneValidator.ZoneValid(project, zone); err != nil {
+			return "", err
+		}
+		return zone, nil
+	}
+	if !p.metadataClient.OnGCE() {
+		return "", fmt.Errorf("zone cannot be determined because build is not running on GCE")
+	}
+	// determine zone based on the zone Cloud Build is running in
+	zone, err = p.metadataClient.Zone()
+	if err != nil || zone == "" {
+		return "", fmt.Errorf("can't infer zone: %v", err)
+	}
+	return zone, nil
+}
+
+func (p *ParamValidatorAndPopulator) determineScratchBucket(originalBucket, project, region, buildID string) (scratchBucket string, err error) {
+	if originalBucket != "" {
+		scratchBucket = originalBucket
+	} else {
+		safeProjectName := strings.Replace(project, "google", "elgoog", -1)
+		safeProjectName = strings.Replace(safeProjectName, ":", "-", -1)
+		if strings.HasPrefix(safeProjectName, "goog") {
+			safeProjectName = strings.Replace(safeProjectName, "goog", "ggoo", 1)
+		}
+		bucket := strings.ToLower(safeProjectName + "-ovf-import-bkt-" + region)
+		it := p.bucketIteratorCreator.CreateBucketIterator(context.Background(), p.storageClient, project)
+		var projectHasBucket bool
+		for itBucketAttrs, err := it.Next(); err != iterator.Done; itBucketAttrs, err = it.Next() {
+			if err != nil {
+				return "", err
+			}
+			if itBucketAttrs.Name == bucket {
+				projectHasBucket = true
+				break
+			}
+		}
+		if !projectHasBucket {
+			p.logger.User(fmt.Sprintf("Creating scratch bucket `%v` in %v region", bucket, region))
+			if err := p.storageClient.CreateBucket(
+				bucket, project,
+				&storage.BucketAttrs{Name: bucket, Location: region}); err != nil {
+				return "", err
+			}
+		}
+		scratchBucket = fmt.Sprintf("gs://%v/", bucket)
+	}
+
+	return pathutils.JoinURL(scratchBucket, buildID), nil
+}
+
+func (p *ParamValidatorAndPopulator) determineReleaseTrack(releaseTrack string) (string, error) {
+	if releaseTrack == "" {
+		releaseTrack = ovfdomain.GA
+	}
+	if releaseTrack != ovfdomain.Alpha && releaseTrack != ovfdomain.Beta && releaseTrack != ovfdomain.GA {
+		return "", daisy.Errf("invalid value for release-track flag: %v", releaseTrack)
+	}
+	return releaseTrack, nil
+}
+
+func (p *ParamValidatorAndPopulator) determineDeadline(originalTimeout string) (time.Time, error) {
+	timeout, err := time.ParseDuration(originalTimeout)
+	if err != nil {
+		return time.Now(), daisy.Errf("Error parsing timeout `%v`", originalTimeout)
+	}
+	return time.Now().Add(timeout), nil
+}
+
+func (p *ParamValidatorAndPopulator) determineBuildID(originalTimeout string) string {
+	if originalTimeout != "" {
+		return originalTimeout
+	}
```

Comment:
`originalTimeout` seems like a copy/paste error here? Should be `buildIDFlagValue` or similar.

Reply:
shoot, good catch, done.